src/odb/src/db/dbInsertBuffer.cpp

```diff
@@ -0,0 +1,1685 @@
+
+#include "dbInsertBuffer.h"
+
+#include <cassert>
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <set>
+#include <string>
+#include <string_view>
+
+#include "dbInst.h"
+#include "dbModule.h"
+#include "dbNet.h"
+#include "odb/db.h"
+#include "odb/dbObject.h"
+#include "odb/dbTypes.h"
+#include "odb/geom.h"
+#include "utl/Logger.h"
+
+namespace odb {
+
+dbInsertBuffer::dbInsertBuffer(dbNet* net)
+    : net_(net),
+      block_(net ? net->getBlock() : nullptr),
+      logger_(net ? net->getImpl()->getLogger() : nullptr)
+{
+  assert(net_ != nullptr);
+  assert(block_ != nullptr);
+  assert(logger_ != nullptr);
+}
+
+void dbInsertBuffer::resetMembers()
+{
+  buf_input_iterm_ = nullptr;
+  buf_output_iterm_ = nullptr;
+  new_flat_net_ = nullptr;
+  new_mod_net_ = nullptr;
+  target_module_ = nullptr;
+  buffer_master_ = nullptr;
+  new_buf_base_name_ = nullptr;
+  new_net_base_name_ = nullptr;
+  uniquify_ = dbNameUniquifyType::ALWAYS;
+  orig_drvr_pin_ = nullptr;
+  is_target_only_cache_.clear();
+  module_reusable_nets_.clear();
+}
+
+dbInst* dbInsertBuffer::insertBufferBeforeLoad(
+    dbObject* load_input_term,
+    const dbMaster* buffer_master,
+    const Point* loc,
+    const char* new_buf_base_name,
+    const char* new_net_base_name,
+    const dbNameUniquifyType& uniquify)
+{
+  dlogInsertBufferStart(++insert_buffer_call_count_, "BeforeLoad");
+
+  return insertBufferSimple(load_input_term,
+                            buffer_master,
+                            loc,
+                            new_buf_base_name,
+                            new_net_base_name,
+                            uniquify,
+                            /* insertBefore */ true);
+}
+
+dbInst* dbInsertBuffer::insertBufferAfterDriver(
+    dbObject* drvr_output_term,
+    const dbMaster* buffer_master,
+    const Point* loc,
+    const char* new_buf_base_name,
+    const char* new_net_base_name,
+    const dbNameUniquifyType& uniquify)
+{
+  dlogInsertBufferStart(++insert_buffer_call_count_, "AfterDriver");
+
+  return insertBufferSimple(drvr_output_term,
+                            buffer_master,
+                            loc,
+                            new_buf_base_name,
+                            new_net_base_name,
+                            uniquify,
+                            /* insertBefore */ false);
+}
+
+dbInst* dbInsertBuffer::insertBufferSimple(dbObject* term_obj,
+                                           const dbMaster* buffer_master,
+                                           const Point* loc,
+                                           const char* new_buf_base_name,
+                                           const char* new_net_base_name,
+                                           const dbNameUniquifyType& uniquify,
+                                           bool insertBefore)
+{
+  if (term_obj == nullptr || buffer_master == nullptr) {
+    return nullptr;
+  }
```

Comment:
These seems more like an error by the caller.  I think error here would be better than a quiet nullptr.